### ISSUE TITLE
cudaGetLastError() results in infinite loop

### ISSUE THREAD

**Issue by fwyzard** [ComputationalRadiationPhysics/cupla#138]
Calling `cudaGetLastError()` results while using Cupla in header-only mode in an infinite loop.

#### `test.cc`
```c++
#include <iostream>

#include <cupla/config/GpuCudaRt.hpp>

int main(void) {
  auto status = cudaGetLastError();

  std::cout << "status: " << status << std::endl;

  return 0;
}
```

Build and run with
```bash
nvcc -x cu -std=c++11 -O2 -g -I/usr/local/alpaka/alpaka/include -I/usr/local/alpaka/cupla/include -w test.cc -o test
./test
```

The program gets stuck on line 6, and it never reaches the `std::cout` statement.

Removing the `#include <cupla/config/GpuCudaRt.hpp>` and running with CUDA natively works, as expected.

**Comment by fwyzard** [comment]
I forgot to add that I am using the `devel` branch of Alpaka and the `dev` branch of Cupla.

**Comment by fwyzard** [comment]
From a quick look at the code, I guess the reason is that in [include/cupla/cudaToCupla/runtime.hpp](https://github.com/ComputationalRadiationPhysics/cupla/blob/dev/include/cupla/cudaToCupla/runtime.hpp#L64) `cudaGetLastError()` is `#define`d to `cuplaGetLastError()`:
```c++
#define cudaGetLastError(...) cuplaGetLastError(__VA_ARGS__)
```

and in [src/common.cpp](https://github.com/ComputationalRadiationPhysics/cupla/blob/dev/src/common.cpp#L47-L59) `cuplaGetLastError()` calls `cudaGetLastError()`:
```c++
CUPLA_HEADER_ONLY_FUNC_SPEC
cuplaError_t
cuplaGetLastError()
{
#if( ALPAKA_ACC_GPU_CUDA_ENABLED == 1 )
    // reset the last cuda error
    return (cuplaError_t)cudaGetLastError();
#elif( ALPAKA_ACC_GPU_HIP_ENABLED == 1 )
    return (cuplaError_t)hipGetLastError();
#else
    return cuplaSuccess;
#endif
}
```

which means it class itself indefinitely.

**Comment by fwyzard** [comment]
@waredjeb FYI

**Comment by psychocoderHPC** [comment]
thanks for the report, the bug is fixed with #139.

**Comment by ax3l** [comment]
Confirmed [here](https://github.com/ComputationalRadiationPhysics/cupla/pull/139#issuecomment-570778076), thank you for the detailed report! :)
